spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/elasticsearch/ElasticsearchRestClientAutoConfigurationTests.java

```diff
@@ -220,6 +226,37 @@ void configureUriWithUsernameAndPasswordWhenUsernameAndPasswordPropertiesSet() {
 				});
 	}
 
+	@Test
+	void configureShouldOnlyCreateSnifferInstance() {
+		this.contextRunner.run(
+				(context) -> assertThat(context).doesNotHaveBean(RestClient.class)
+				.hasSingleBean(RestHighLevelClient.class));
+		this.contextRunner.run(
+				(context) -> assertThat(context).hasSingleBean(Sniffer.class));
+	}
+
+	@Test
+	void configureShouldHaveSnifferInstance() {
+		this.contextRunner.run(
+				(context) -> assertThat(context).doesNotHaveBean(RestClient.class)
+						.hasSingleBean(RestHighLevelClient.class));
+		this.contextRunner.run(
+				(context) -> {
+					assertThat(context).hasSingleBean(Sniffer.class);
+					Sniffer sniffer = context.getBean(Sniffer.class);
+					Assert.assertNotNull(sniffer);
+				});
+	}
+
+	@Test
+	void configureWithCustomSetIntervalProperties() {
+		this.contextRunner.withPropertyValues("spring.elasticsearch.sniff.sniffInterval=15s, " +
+				"spring.elasticsearch.sniff.sniffFailureDelay=15s").run((context) -> {
+			Sniffer sniffer = context.getBean(Sniffer.class);
+			Assert.assertNotNull(sniffer);
+			});
+	}
+
```

Comment:
Two tests are missing here.

One that assert what happens when the library is not on the classpath, as I've indicated in [my previous review](https://github.com/spring-projects/spring-boot/pull/24340#discussion_r546777503).

One that assert that a custom `Sniffer` instance is used rather creating one here. This custom instance should probably have a dependency on the high level client we auto-configure to make this a bit more realistic.